app/src/main/java/com/github/sdpsharelook/textDetection/TextDetection.kt

```diff
@@ -0,0 +1,36 @@
+package com.github.sdpsharelook.textDetection
+
+import com.google.android.gms.tasks.OnFailureListener
+import com.google.android.gms.tasks.OnSuccessListener
+import com.google.mlkit.vision.common.InputImage
+import com.google.mlkit.vision.text.Text
+import com.google.mlkit.vision.text.TextRecognition
+import com.google.mlkit.vision.text.latin.TextRecognizerOptions
+
+class TextDetection {
+    private val recognizer = TextRecognition.getClient(TextRecognizerOptions.DEFAULT_OPTIONS)
+
+    fun process(image: InputImage): String{
+        var textResult = "Aucun text"
+        if(recognizer.process(image)
+            .addOnSuccessListener(
+                OnSuccessListener<Text?> { texts ->
+                    textResult = texts.text
+                })
+            .addOnFailureListener(
+                OnFailureListener { e -> // Task failed with an exception
+                    e.printStackTrace()
+                }).isComplete){
+            return textResult
+        }
+        return textResult

```

Comment:
very bad handling of asynchronous code: the task is likely to never be complete
make your function `suspend` and use `Task.await()`